src/response.ts

```diff
@@ -127,25 +141,32 @@ document.addEventListener('DOMContentLoaded', async () => {
   window.addEventListener(
     'message',
     async function (e) {
-      // Addresses must be static
-      if (e.origin !== 'http://localhost:7071' || e.data !== '3DS.Notification.Received') {
-        return;
-      } else {
-        debug('MESSAGE RECEIVED: ', e.data);
-        await getTransactionFromSessionStorageTask('payment')
-          .chain(transaction =>
-            getStringFromSessionStorageTask('sessionToken')
-              .chain(sessionToken => resumeTransactionTask('Y', sessionToken, transaction.token, pmClient))
-              .chain(_ => checkStatusTask(transaction.token, paymentManagerClientWithPollingOnPreAcs)),
-          )
-
-          .fold(
-            _ => debug('To handle error'),
-            transactionStatus =>
-              start3DS2AcsChallengeStep(transactionStatus.data.acsUrl, transactionStatus.data.params, document.body),
-          )
-          .run();
-      }
+      await fromPredicate<Error, MessageEvent<any>>(
+        e1 => e1.origin === 'http://localhost:7071' && e1.data === '3DS.Notification.Received',
+        toError,
+      )(e)
+        .fold(
+          _ => null, // TODO error handle
```

Comment:
When we'll merge this [PR](https://github.com/pagopa/io-pay/pull/49) remember to put here `THREEDSMETHODURL_STEP1_RESP_ERR`